### DISCUSSION TITLE
Which OSG version are you using ?

### DISCUSSION THREAD

**Discussion by VirtualSim** [pelicanmapping/osgearth#2404] [Support Q&A]
Hi community,
I would like to know which version of OSG are you using with osgEarth 3.5, for Windows, in 64bits ? (and the VisualStudio version used please).
I am trying to get an Osg 3.6.5 under GL3 to just display things (it does not as soon as I switch GL3 Available, complaining about OpenGL unsupported call) and the binary version I have is GL2 only (so, not compatible with osgEarth 3.5)
I think that most of you are on Linux ;-)
Thanks

**Comment by remoe** [comment]
Do you have read the docs ?

https://github.com/gwaldron/osgearth/blob/master/docs%2Fsource%2Fbuild.md

OSG 3.6.5 works. Try to use GL Core.
And one can read the github actions.

**Reply by keince** [reply]
For the record, I have read the docs at the link provided and have followed the suggestions in step 2. I still don't see GL3 being used in my build of osgEarth 3.5.

Below is my output when I run osgearth_viewer in my build...

[osgearth info] [Capabilities] osgEarth Version:  3.5.0 build 149
[osgearth info] [Capabilities] GDAL Version:      3.7.2
[osgearth info] [Capabilities] OSG Version:       3.6.5
[osgearth info] [Capabilities] OSG GL3 Features:  no
[osgearth info] [Capabilities] OSG FFP Available: yes
[osgearth info] [Capabilities] CPU Cores:         12
[osgearth info] [Capabilities] GL_VENDOR:         NVIDIA Corporation
[osgearth info] [Capabilities] GL_RENDERER:       GeForce GTX 1660 Ti/PCIe/SSE2
[osgearth info] [Capabilities] GL_VERSION:        4.6.0 NVIDIA 462.75
[osgearth info] [Capabilities] GL CORE Profile:   no

This suggest that GL3 features are still not enabled...

**Comment by gwaldron** [comment]
You *can* run osgEarth under the GL2 (default) OSG profile. Check out the FAQ on the topic here:

https://docs.osgearth.org/en/latest/faq.html#error-gl-modelviewmatrix-is-removed-after-version-140

If you want to use GL3 you need to build OSG itself in either the GL3 or GLCORE profile.

If you are building OSG manually, you need to set the OSG CMAKE option `OPENGL_PROFILE` to either `GL3` or `GLCORE`. The OSG CMAKE system will then set up all the other necessary flags for you. *Do not attempt to set flags such as OSG_GL3_AVAILABLE by hand as this will screw things up.*

If you are building OSG from `vcpkg` you need to tell OSG to use the `GL3` or `GLCORE` profile in a different way, by editing the OSG portfilethat comes with `vcpkg`.  There is a comment in there telling you how to do it.

Hope this helps.
